test/data_stream_test.rb

```diff
@@ -548,14 +548,14 @@ module Pocolog
                               [base_time + 1, base_time + 11, 1]], stream.each.to_a
             end
 
-            it "removes a sample whose time strictly equals the given time" do
-                stream = @stream.to_logical_time(base_time + 12)
+            it "keeps a sample whose time strictly equals the given time" do
+                stream = @stream.to_logical_time(base_time + 11)
                 assert_equal [[base_time + 0, base_time + 10, 0],
                               [base_time + 1, base_time + 11, 1]], stream.each.to_a
             end
 
             it "updates the stream index of the returned stream" do
```

Comment:
I don't understand how it is testing this: "updates the stream index of the returned stream"; if a time passed the last is removed, all indexes should remain the same right? Also, there isn't a direct check for the index

Reply:
Right.

The `index` is a data structure internal to the stream that is used when seeking. So ... it's essentially making sure that the index is not corrupted.

But I agree with you ... while this way of testing the update kind-of made sense for from_logical_time, it really does not for to_logical_time (since the seek will work either way)

Updated to check the start time (for from_logical_time) and end time (for to_logical_time) of the index.